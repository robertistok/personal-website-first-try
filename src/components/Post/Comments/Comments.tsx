import React from "react";
import ReactDisqusComments from "react-disqus-comments";
import { useSiteMetadata } from "../../../hooks";

interface CommentsProps {
  postTitle?: string;
  postSlug?: string;
}

const Comments: React.FunctionComponent<CommentsProps> = ({
  postTitle,
  postSlug
}: CommentsProps): React.ReactElement => {
  const { url, disqusShortname } = useSiteMetadata();

  if (!disqusShortname) {
    return null;
  }

  return (
    <ReactDisqusComments
      shortname={disqusShortname}
      identifier={postTitle}
      title={postTitle}
      url={url + postSlug}
    />
  );
};

export default Comments;
