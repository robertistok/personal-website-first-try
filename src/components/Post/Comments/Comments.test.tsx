import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { useStaticQuery, StaticQuery } from "gatsby";

import Comments from "./Comments";
import siteMetadata from "../../../../jest/__fixtures__/site-metadata";

describe.only("<Comments />", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockReturnValue(siteMetadata);
  });

  const props = {
    postTitle: "test",
    postSlug: "/test"
  };

  it("renders", () => {
    shallow(<Comments {...props} />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<Comments {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
