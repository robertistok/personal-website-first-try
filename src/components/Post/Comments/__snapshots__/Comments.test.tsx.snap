// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> renders and matches snapshot 1`] = `
<DisqusThread
  identifier="test"
  shortname="Test disquesShortname"
  title="test"
  url="http://localhost/test"
/>
`;
